nemo/collections/nlp/data/machine_translation/preproc_mt_data.py

```diff
@@ -366,6 +348,91 @@ def preprocess_parallel_dataset(
 
         return tar_file_paths, metadata_path
 
+    @staticmethod
+    def _get_num_lines(filename):
+        with open(filename) as f:
+            for i, l in enumerate(f):
+                pass
+        return i + 1
+
+    @staticmethod
+    def _get_lines_partition(num_lines, lines_per_dataset_fragment):
+        # create partition based on fragment size
+        fragment_indices = []
+        for i in range(0, num_lines, lines_per_dataset_fragment):
+            fragment_indices.append([i, i + lines_per_dataset_fragment])
+        # remove last indices if needed
+        if fragment_indices[-1][1] >= num_lines:
```

Comment:
For large values of lines_per_dataset_fragment, this will potentially result in a lot of lines lost right? I think we should still process this fragment and just throw away batches in `_process_fragment()`

Reply:
Good point, I can make that change.

Reply:
Done.